shell/browser/native_window_mac.mm

```diff
@@ -597,28 +597,16 @@ void ViewDidMoveToSuperview(NSView* self, SEL _cmd) {
 }
 
 void NativeWindowMac::Close() {
-  // When this is a sheet showing, performClose won't work.
-  if (is_modal() && parent() && IsVisible()) {
-    NSWindow* window = parent()->GetNativeWindow().GetNativeNSWindow();
-    if (NSWindow* sheetParent = [window sheetParent]) {
-      base::ThreadTaskRunnerHandle::Get()->PostTask(
-          FROM_HERE, base::BindOnce(base::RetainBlock(^{
-            [sheetParent endSheet:window];
-          })));
-    }
-
-    // Manually emit close event (not triggered from close fn)
-    NotifyWindowCloseButtonClicked();
-    CloseImmediately();
-    return;
-  }
-
   if (!IsClosable()) {
     WindowList::WindowCloseCancelled(this);
     return;
   }
 
   [window_ performClose:nil];
+
+  if (is_modal() && parent() && IsVisible()) {
```

Comment:
Should add a comment on why this is needed, here and in `electron_ns_window_delegate.mm`.